packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -63,7 +69,7 @@ export class AwsCliCompatible {
       return makeCachingProvider(fromIni({
         profile: options.profile,
         ignoreCache: true,
-        mfaCodeProvider: tokenCodeFn,
+        mfaCodeProvider: makeTokenCodeFn(this.ioHelper),
```

Comment:
`tokenCodeFn` could probably be a member function, then it would have access to `this.ioHelper` already.